src/pages/purchase/kinguin.tsx

```diff
@@ -0,0 +1,216 @@
+import { useEffect, useRef, useState } from 'react';
+import { useRouter } from 'next/router';
+import { Container, Center, Loader, Text, Alert } from '@mantine/core';
+import { IconAlertCircle } from '@tabler/icons-react';
+import { createServerSideProps } from '~/server/utils/server-side-helpers';
+import { Meta } from '~/components/Meta/Meta';
+import Script from 'next/script';
+
+declare global {
+  interface Window {
+    kinguinCheckoutSDK?: {
+      init: (config: KinguinCheckoutConfig) => void;
+    };
+  }
+}
+
+interface KinguinCheckoutConfig {
+  wrapperSelector: string;
+  productsSelector: string;
+  productUrlAttribute: string;
+  iframeId: string;
+  iframeWidth: string;
+  iframeHeight: string;
+  popupWidth?: number;
+  popupHeight?: number;
+  translations?: {
+    popupOpenFail?: string;
+  };
+  discount?: string;
+  language?: string;
+  currency?: string;
+  email?: string;
+}
+
+export default function KinguinPurchasePage() {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [sdkLoaded, setSdkLoaded] = useState(false);
+  const checkoutInitialized = useRef(false);
+
+  const { productId, productType, productName } = router.query as {
+    productId?: string;
+    productType?: string;
+    productName?: string;
+  };
+
+  const handleSDKLoad = () => {
+    console.log('Kinguin SDK loaded');
+    setSdkLoaded(true);
+  };
+
+  // Build the Kinguin product URL
+  const buildKinguinUrl = (productId: string) => {
+    return `https://www.kinguin.net/category/${productId}/civitai-gift-card?referrer=civitai.com`;
+  };
+
+  const initializeCheckout = () => {
+    console.log('Initializing Kinguin checkout...');
+    if (!window.kinguinCheckoutSDK || !productId || checkoutInitialized.current) {
+      console.error('Kinguin SDK not available or already initialized');
+      return;
+    }
+
+    checkoutInitialized.current = true;
+
+    const config: KinguinCheckoutConfig = {
+      wrapperSelector: '#kinguin-checkout-wrapper',
+      productsSelector: '.kinguin-product-button',
+      productUrlAttribute: 'data-product-url',
+      iframeId: 'kinguin-checkout-iframe',
+      iframeWidth: '100%',
+      iframeHeight: '700px',
+      popupWidth: 900,
+      popupHeight: 1000,
+      translations: {
+        popupOpenFail: 'Window could not be opened, please try again.',
+      },
+      language: 'en',
+      currency: 'USD',
+    };
```

Comment:
Could easily live outside of the component 🤷 just so it doesn't get instantiated every time this function gets called